server_addon/create_ayon_rez_addons.py

```diff
@@ -0,0 +1,205 @@
+import os
+import sys
+import re
+import json
+import shutil
+import argparse
+import zipfile
+import platform
+import collections
+from pathlib import Path
+from typing import Any, Optional, Iterable, Pattern, List, Tuple
+
+
+def read_addon_version(version_path: Path) -> str:
+    # Read version
+    version_content: dict[str, Any] = {}
+    with open(str(version_path), "r") as stream:
+        exec(stream.read(), version_content)
+    return version_content["__version__"]
+
+
+def create_addon_package(
+    addon_dir: Path,
+    output_dir: Path,
+    create_zip: bool,
+    keep_sources: bool
+):
+    addon_name = addon_dir.name
+    openpype_dir = None
+
+    if addon_name == "openpype":
+        openpype_dir = addon_dir / ".." / ".." / "openpype"
+        version_path = openpype_dir / "version.py"
+    else:
+        version_path = addon_dir / "server" / "version.py"
+
+    addon_version = read_addon_version(version_path)
+
+    addon_output_dir = output_dir / addon_dir.name / addon_version
+
+    if addon_output_dir.exists():
+        shutil.rmtree(str(addon_output_dir))
+
+    addon_output_dir.mkdir(parents=True)
+
+    shutil.copytree(
+        addon_dir,
+        addon_output_dir,
+        ignore=shutil.ignore_patterns(
+            '*.pyc',
+            '.*',
+            '*__pycache__*',
+        ),
+        dirs_exist_ok=True,
+    )
+
+    package_py = addon_output_dir / "package.py"
+
+    with open(package_py, 'w+') as pkg_py:
+        pkg_py.write(
+f"""name = "{addon_dir.name}"
+version = "{addon_version}"
+plugin_for = ["ayon_server"]
+build_command = "python {{root}}/rezbuild.py"
+"""
+        )
+
+    if addon_dir.name == "openpype":
+        ignored_hosts = []
```

Comment:
Ignored hosts are outdated from the other script. Also there are folders in this root that contain modules/hosts that should be ignored, which is not handled in this logic.